### PR TITLE
test: add NewFromMap and Validate func

### LINKED ISSUE #416: Add unit tests to config util functions
The task is to add unit tests to the `validate()` and `newFromMap()` functions in [`/pkg/util/config/config.go`](https://github.com/zalando-incubator/postgres-operator/blob/master/pkg/util/config/config.go)

### PR DESCRIPTION
**Pull request by ZYecho**
Signed-off-by: zhangyue <zy675793960@yeah.net>
as the title describe, also change some inappropriate code style.
fix #416

### CODE REVIEW

**Comment by Jan-M** [comment]
Thanks for your PR, looks good to me.

@sdudoladov

**Comment by erthalion** [comment]
@ZYecho  Thanks for the PR. Unfortunately, as you can see from Travis build, test doesn't pass:
```
Unable to detect operator namespace from within its pod due to:
open /var/run/secrets/kubernetes.io/serviceaccount/namespace: no such 
```

It happens because operator has `oauth_token_secret_name` in the config and tries to resolve namespace for it. Probably you can either remove it from the config, or somehow mock namespace functionality.

**Comment by ZYecho** [comment]
> @ZYecho Thanks for the PR. Unfortunately, as you can see from Travis build, test doesn't pass:
> 
> ```
> Unable to detect operator namespace from within its pod due to:
> open /var/run/secrets/kubernetes.io/serviceaccount/namespace: no such 
> ```
> It happens because operator has `oauth_token_secret_name` in the config and tries to resolve namespace for it. Probably you can either remove it from the config, or somehow mock namespace functionality.

you mean 
`oauth_token_secret_name` not need now in the `config` struct?

**Comment by erthalion** [comment]
> you mean `oauth_token_secret_name` not need now in the `config` struct?

I mean you can try to avoid it to reduce the scope of the test, or mock namespace functionality - otherwise you're test is always red. Please note, that a config values in `NewFromMap` are taken via reflection, taking into account corresponding default values - so you're not just passing them inside via function arguments.

**Comment by ZYecho** [comment]
> > you mean `oauth_token_secret_name` not need now in the `config` struct?
> 
> I mean you can try to avoid it to reduce the scope of the test, or mock namespace functionality - otherwise you're test is always red. Please note, that a config values in `NewFromMap` are taken via reflection, taking into account corresponding default values - so you're not just passing them inside via function arguments.

I change the test scope to avoid it, PTAL, thanks.

File: pkg/spec/types.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ type syncUserOperation int
 // Possible values for the sync user operation (removal of users is not supported yet)
 const (
 	PGSyncUserAdd = iota
-	PGsyncUserAlter
+	PGSyncUserAlter
```
Comment (erthalion) [inline pkg/spec/types.go]: Why do you need this change, it seems unrelated, no?

File: pkg/util/config/config.go [hunk 2]
```diff
@@ -163,13 +163,13 @@ func Copy(c *Config) Config {
 	return cfg
 }
 
-func validate(cfg *Config) (err error) {
+func Validate(cfg *Config) error {
```
Comment (erthalion) [inline pkg/util/config/config.go]: Why to capitalize this and `ProcessField`? They're in the same package with the tests, so as far as I remember should be visible anyway.

File: pkg/util/config/util_test.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+		mapFields[field.Name] = field
+	}
+
+	tests := []struct {
+		name    string
+		args    map[string]string
+		wantErr bool
+	}{
+		{
+			name: "AllFields-ok",
```
Comment (erthalion) [inline pkg/util/config/util_test.go]: Probably it's a good idea to test `validate` and `processField` separately, but then you test pretty small part of `processField` as far as I see. Maybe it makes sense to get all the possible fields from an empty config via `structFields` (in  this way we also can test it) and then just exclude "inconvenient for testing" fields, that will require some mocking to test them.

**Comment by erthalion** [comment]
@ZYecho overall looks better, thanks. But I have few commentaries/questions, please take a look at the review.
